### PR TITLE
Fix failing PHP tests

### PR DESCRIPTION
**Pull request by samnajian**
## Description
- Our test suit had issues running MariaDB with `mariadb:latest` and this PR pins the version number to the latest working version.
- Makes sure the pnpm version used matches the pinned version in WC plugin.

### CODE REVIEW

File: .github/workflows/php_tests.yml [hunk 1]
```diff
@@ -62,13 +62,17 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: '8.2'
+      - name: Get WC pinned pnpm version
+        id: get_wc_pnpm_version
+        run: echo "WC_PNPM=$(node -p "require('/tmp/woocommerce/plugins/woocommerce/package.json').engines.pnpm")"  >> $GITHUB_OUTPUT
```
Comment (waclawjacek) [inline .github/workflows/php_tests.yml]: This works but I think the quotes should be escaped. What's happening right now is this (string concatenation):

```
waclaw@Waclaws-MacBook-Pro woocommerce % echo "hey"foo"hey"
heyfoohey
```

How about this instead?

```suggestion
        run: echo "WC_PNPM=$(node -p \"require('/tmp/woocommerce/plugins/woocommerce/package.json').engines.pnpm\")" >> "$GITHUB_OUTPUT"
```

**Review by waclawjacek — CHANGES_REQUESTED** [review]
What a great solution! Suggesting what I think is an improvement (adding some escaping).

**Comment by samnajian** [comment]
@waclawjacek The applied suggestion didn't work, not sure what I should do with it, would it be fine to revert the last commit?

File: .github/workflows/php_tests.yml [hunk 2]
```diff
@@ -62,13 +62,17 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: '8.2'
+      - name: Get WC pinned pnpm version
+        id: get_wc_pnpm_version
+        run: echo "WC_PNPM=$(node -p \"require('/tmp/woocommerce/plugins/woocommerce/package.json').engines.pnpm\")" >> "$GITHUB_OUTPUT"
```
Comment (waclawjacek) [inline .github/workflows/php_tests.yml]: Sorry, my bad! I've just learned that using `$()` creates a new quoting context. So this should work (it does when I test it locally after defining `GITHUB_OUTPUT`):

```suggestion
        run: echo "WC_PNPM=$(node -p "require('/tmp/woocommerce/plugins/woocommerce/package.json').engines.pnpm")" >> "$GITHUB_OUTPUT"
```

**Review by waclawjacek — CHANGES_REQUESTED** [review]

**Review by waclawjacek — APPROVED** [review]

**Pull request merged**